src/expression.c

```diff
@@ -13727,9 +13728,92 @@ Expression *CondExp::semantic(Scope *sc)
     printf("e1 : %s\n", e1->type->toChars());
     printf("e2 : %s\n", e2->type->toChars());
 #endif
+
+    /* Bugzilla 14696: If either e1 or e2 contain temporaries which need dtor,
+     * make them conditional.
+     * Rewrite:
+     *      cond ? (__tmp1 = ..., __tmp1) : (__tmp2 = ..., __tmp2)
+     * to:
+     *      (auto __cond = cond) ? (... __tmp1) : (... __tmp2)
+     * and replace edtors of __tmp1 and __tmp2 with:
+     *      __tmp1->edtor --> __cond && __tmp1.dtor()
+     *      __tmp2->edtor --> __cond || __tmp2.dtor()
+     */
+    hookDtors(sc);
+
     return this;
 }
 
+void CondExp::hookDtors(Scope *sc)
+{
+    class DtorVisitor : public StoppableVisitor
+    {
+    public:
+        Scope *sc;
+        CondExp *ce;
+        VarDeclaration *vcond;
+        bool isThen;
+
+        DtorVisitor(Scope *sc, CondExp *ce)
+        {
+            this->sc = sc;
+            this->ce = ce;
+            this->vcond = NULL;
+        }
+
+        void visit(Expression *e)
+        {
+            //printf("(e = %s)\n", e->toChars());
+        }
+
+        void visit(DeclarationExp *e)
+        {
+            VarDeclaration *v = e->declaration->isVarDeclaration();
+            if (v && !v->noscope && !v->isDataseg())
```

Comment:
Looks like `noscope` could readily be used for a [destructive move](https://issues.dlang.org/show_bug.cgi?id=14466) intrinsic.